Parse/ParseObject.cs

```diff
@@ -286,65 +273,18 @@ string propertyName
     internal virtual void SetDefaultValues() {
     }
 
-    internal static string GetClassName(Type t) {
-      var attr = t.GetTypeInfo().GetCustomAttribute<ParseClassNameAttribute>();
-      if (attr == null) {
-        throw new ArgumentException("No ParseClassName attribute specified on the given subclass.");
-      }
-      return attr.ClassName;
-    }
-
-    /// <summary>
-    /// Gets the appropriate factory for the given class name. If there is no factory for the class,
-    /// a factory that produces a regular ParseObject will be created.
-    /// </summary>
-    /// <param name="className">The class name for the ParseObjects the factory will create.</param>
-    /// <returns></returns>
-    private static Func<ParseObject> GetFactory(string className) {
-      Tuple<Func<ParseObject>, Type> result;
-      if (!objectFactories.TryGetValue(className, out result)) {
-        return () => new ParseObject(className);
-      }
-      return result.Item1;
-    }
-
     /// <summary>
     /// Registers a custom subclass type with the Parse SDK, enabling strong-typing of those ParseObjects whenever
     /// they appear. Subclasses must specify the ParseClassName attribute, have a default constructor, and properties
     /// backed by ParseObject fields should have ParseFieldName attributes supplied.
     /// </summary>
     /// <typeparam name="T">The ParseObject subclass type to register.</typeparam>
-    public static void RegisterSubclass<T>() where T : ParseObject, new() {
-      var className = GetClassName(typeof(T));
-      if (className == null) {
-        throw new ArgumentException("No ParseClassName attribute defined for " + typeof(T));
-      }
-
-      Tuple<Func<ParseObject>, Type> oldValue;
-      if (objectFactories.TryGetValue(className, out oldValue)) {
-        if (typeof(T).GetTypeInfo().IsAssignableFrom(oldValue.Item2.GetTypeInfo())) {
-          // The old class was already more descendant than the new subclass type. No-op.
-          return;
-        }
-        if (className.Equals(GetClassName(typeof(ParseUser)))) {
-          ParseUser.ClearInMemoryUser();
```

Comment:
What's the reason to not put it inside `SubclassingController.RegisterSubclass(typeof(T));`? If you don't like the fact that we're using `ParseUser` and `ParseInstallation` inside `SubclassingController`, we can inject `IParseCurrentUserController` and `IParseCurrentInstallationController` into `SubclassingController` and clear the memory from there.